Public/Src/Engine/Processes/SandboxedProcessReports.cs

```diff
@@ -522,7 +529,77 @@ private void AddLookupEntryForProcessExit(uint processId, ReportedProcess report
             m_processesExits[processId] = reportedProcess;
         }
 
-        private bool FileAccessReportLineReceived<T>(ref T data, FileAccessReportProvider<T> parser, out string errorMessage)
+        private bool TryParseAugmentedFileAccess(
+            ref string data,
+            out uint processId,
+            out ReportedFileOperation operation,
+            out RequestedAccess requestedAccess,
+            out FileAccessStatus status,
+            out bool explicitlyReported,
+            out uint error,
+            out Usn usn,
+            out DesiredAccess desiredAccess,
+            out ShareMode shareMode,
+            out CreationDisposition creationDisposition,
+            out FlagsAndAttributes flagsAndAttributes,
+            out AbsolutePath manifestPath,
+            out string path,
+            out string enumeratePattern,
+            out string processArgs,
+            out string errorMessage)
+        {
+            // An augmented file access has the same structure as a regular one, so let's call
+            // the usual parser
+            var result = FileAccessReportLine.TryParse(
+                ref data, 
+                out processId, 
+                out operation, 
+                out requestedAccess, 
+                out status, 
+                out explicitlyReported, 
+                out error, 
+                out usn, 
+                out desiredAccess, 
+                out shareMode, 
+                out creationDisposition, 
+                out flagsAndAttributes, 
+                out manifestPath, 
+                out path, 
+                out enumeratePattern, 
+                out processArgs, 
+                out errorMessage);
+

```

Comment:
don't you have to check if `result` is `true` before proceeding?

Reply:
I could, but It's not really needed, since here I'm just forwarding the result of parsing to the caller. The post-processing that you see after FileAccessReportLine.TryParse(...) is only acting when the returned path can be interpreted as an absolute path. That always make sense even if result is false.

---
In reply to: [343778501](https://github.com/microsoft/BuildXL/pull/1175#discussion_r343778501) [](ancestors = 343778501)